Recon/include/Recon/OverlayProducer.h

```diff
@@ -65,6 +65,9 @@ class OverlayProducer : public framework::Producer {
   void onProcessStart() final override;
 
  private:
+  /// The parameters used to configure this producer
+  framework::config::Parameters params_;
```

Comment:
I _really_ want to avoid persisting the parameters class. It is meant to be the translation layer between the Python and the C++, so it is inherently non-type-specific and thus confusing in the C++ ecosphere. My dream is to have **no** use of Parameters _after_ the python configuration is done being parsed. This would allow us to maintain separation between Python and C++. (There are some complicated SimCore aspects that prevent this currently.)

Reply:
Although I agree in principle, that will require much larger changes that are outside of the scope of this PR.  We should discuss our approach and plan to make these changes towards the end of the year once most of the recon is done and we are able to take time for the software again.